src/libp2p_cache.erl

```diff
@@ -74,8 +74,22 @@ init([TID]) ->
     libp2p_swarm_auxiliary_sup:register_cache(TID),
     SwarmName = libp2p_swarm:name(TID),
     DataDir = libp2p_config:base_dir(TID),
-    Opts = [{file, filename:join([DataDir, SwarmName, "cache.dets"])}],
-    {ok, Dets} = dets:open_file(SwarmName, Opts),
+    CacheFilePath = filename:join([DataDir, SwarmName, "cache.dets"]),
+    Opts = [{file, CacheFilePath}],
+    Open = fun() -> dets:open_file(SwarmName, Opts) end,
+    {ok, Dets} =
+        case Open() of
+            {error, Reason} ->
+                %% Likely corrupted file. Single retry should be fine.
+                lager:warning(
+                    "Failed to open DETS file: ~p, for reason: ~p. Deleting.",
+                    [CacheFilePath, Reason]
+                ),
+                ok = file:delete(CacheFilePath),
```

Comment:
@Vagabond On the second thought - this is bad, since if we get the error for some reason other than corrupt _existing_ file, the deletion will fail, so we need to either go back to handling specific `dets:open_file` error or/and handle|ignore output of `file:delete`.

Reply:
ok, I suppose it will badmatch in a new and exciting way